lib/galaxy/jobs/__init__.py

```diff
@@ -1918,6 +1953,7 @@ def setup_external_metadata(self, exec_dir=None, tmp_dir=None,
         if set_extension:
             for output_dataset_assoc in job.output_datasets:
                 if output_dataset_assoc.dataset.ext == 'auto':
+                    model.StorageMedia.refresh_all_media_credentials(output_dataset_assoc.dataset.dataset.active_storage_media_associations, self.app.authnz_manager, self.sa_session)
```

Comment:
Setting the extension shouldn't require this - right? Something seems off here to me - maybe if I sat with it longer though it would become clear?